lib/libPlugin.lua

```diff
@@ -0,0 +1,724 @@
+}
+
+
+libPlugin.format = dt.new_widget("box"){
+  orientation = "vertical",
+  dt.new_widget("label"){ label = "Format Options" },
+  dt.new_widget("separator"){},
+  libPlugin.format_combobox,
+  libPlugin.jpeg_slider,
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.height = dt.new_widget("entry"){
+  tooltip = "height",
+  text = "0",
+}
+
+libPlugin.height_box = dt.new_widget("box"){    -- TODO: refresh processor combobox
+
+  orientation = "horizontal",
+  dt.new_widget("label"){ label = "Height" },
+  libPlugin.height,
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.width = dt.new_widget("entry"){
+  tooltip = "width",
+  text = "0",
+}
+
+libPlugin.width_box = dt.new_widget("box"){
+  orientation = "horizontal",
+  dt.new_widget("label"){ label = "Width " },
+  libPlugin.width,
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.upscale = dt.new_widget("combobox"){
+  label = "upscale",
+  value = 1, "no", "yes",
+}
+
+libPlugin.global = dt.new_widget("box"){
+  orientation = "vertical",
+  dt.new_widget("label"){ label = "Global Options" },
+  dt.new_widget("separator"){},
+  dt.new_widget("label"){ label = "Max Size" },
+  libPlugin.height_box,
+  libPlugin.width_box,
+  libPlugin.upscale,
+}
+
+-- execute button
+-- TODO: save the values from the various widgets when process is pressed
+
+libPlugin.button = dt.new_widget("button"){
+  label = "Process",
+  clicked_callback = function (_)
+
+    -- get the gui parameters
+    local export_format = libPlugin.format_combobox.value
+
+    -- build the image table
+    local img_table, cnt = libPlugin.build_image_table(dt.gui.action_images, export_format)
+    log.msg(log.debug, "image count is " .. cnt)
+
+    -- make sure there is enough images
+    if plugins[libPlugin.processor_combobox.value].DtPluginMinImages <= cnt then
+
+      -- export the images
+      local success = libPlugin.do_export(img_table, export_format, libPlugin.height.text, libPlugin.width.text, libPlugin.upscale.value)
+
+      -- call the processor
+      log.msg(log.debug, processor_cmds[libPlugin.processor_combobox.value])
+      processor_cmds[libPlugin.processor_combobox.value](img_table, plugins[libPlugin.processor_combobox.value])
+    else
+      log.msg(log.error, "Insufficient images selected, " .. plugins[libPlugin.processor_combobox.value].DtPluginMinImages .. " required")
+    end
+  end
+}
+
+--[[
+  NAME
+    libPlugin.register_processor_lib - build and install the external processor gui
+
+  SYNOPSIS
+    result = dtutils.split_filepath(path_string)
+      filepath - path and filename
+
+  DESCRIPTION
+    split_filepath splits a filepath into the path, filename, basename and filetype and puts
+    that in a table
+
+  RETURN VALUE
+    result - a table containing the path, filename, basename, and filetype
+
+  ERRORS
+
+
+
+]]
+
+function libPlugin.register_processor_lib(name_table)
+
+  log.msg(log.info, "name_table length is ", #name_table)
```

Comment:
that's a pretty cool start. I might consider picking up that code and integrating in the debug-helper lib (unless you come first and mobe debug-helpers to a dtutils.debug library :) )